fitnessllm_dataplatform/utils/logging_utils.py

```
@@ -41,0 +1,131 @@
+        level: str,
+        message: str,
+        data_source: Optional[str] = None,
+        user_id: Optional[str] = None,
+        **kwargs
+    ) -> Dict[str, Any]:
+        """Format log entry with structured data."""
+        log_data = {
+            "level": level,
+            "message": message,
+            "service": self.name,
+            "timestamp": datetime.utcnow().isoformat(),
```

Ruff (0.8.2)

`datetime.datetime.utcnow()` used

(DTZ003)